src/seccompiler/src/seccompiler_bin.rs

```diff
@@ -166,43 +166,67 @@ fn compile(args: &Arguments) -> Result<()> {
     Ok(())
 }
 
-fn main() {
+#[derive(Debug)]
+struct SbExitError {
+    exit_code: ExitCode,
+    error_msg: String,
+}
+
+impl Termination for SbExitError {
+    fn report(self) -> ExitCode {
+        eprintln!("{}", self.error_msg);
+        self.exit_code
+    }
+}
+
+impl SbExitError {
+    fn new(exit_code: ExitCode, error_msg: String) -> Self {
+        SbExitError {
+            exit_code,
+            error_msg,
+        }
+    }
+}
+
+fn main() -> core::result::Result<(), SbExitError> {
```

Comment:
This should return a custom error type (same comment as on `run_with_api` https://github.com/firecracker-microvm/firecracker/pull/3575/commits/f83b7c8cd3fdb3b320ee1fe1f80769c4c498ee56#r1163827837).